### PR TITLE
Replace grpc with http in costUpdater

### PR DESCRIPTION
**Pull request by Qi77Qi**
Part of https://broadinstitute.atlassian.net/browse/GAWB-4074

1. Currently config file is using a test service account, which should be updated once we have a dev project and dev service account(s)

2. This code requires app service account (if we're sharing the same one for different services, but we probably should use different ones)

   2.1 pubsub admin role for pubsub topic (not sure if this can be done by app code) 

   2.2 storage admin role for metadata bucket (this can be done in rawls once we have hamm projects and service accounts)

   Currently, these 2 steps are done mannually

Demo
`sbt costUpdater/run`

```
curl 127.0.0.1:8080/status
{"buildTime":"Fri Feb 22 23:29:30 CST 2019","gitHeadCommit":"d35857231940cb1abca93363db0d0342f275aff2","queueSize":0}% 
```

### CODE REVIEW

**Comment by MatthewBemis** [comment]
for 2.1 in your description, you can add the pubsub admin role for an SA like so: https://github.com/broadinstitute/sam/blob/develop/src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleKeyCacheMonitor.scala#L79

**Comment by Qi77Qi** [comment]
@MatthewBemis  is the code sample sa1 who owns the project setting permission for another sa2? In my case, I only have one sa...I assume same sa can't set iam rules for itself?

File: build.sbt [hunk 1]
```diff
@@ -18,13 +19,32 @@ val protobuf =
...
+lazy val core =
   project
     .in(file("core"))
+     .enablePlugins(BuildInfoPlugin)
     .settings(
-      libraryDependencies ++= Dependencies.automation,
+      libraryDependencies ++= Dependencies.common,
       Settings.commonSettings,
-      Settings.buildInfoSettings
+//    This is not ideal, but BuildInfoPlugin doesn't work as expected for core
```
[the remark this answers is not in the document]
Comment (Qi77Qi) [reply build.sbt]: `costUpdater` is for background process. `server` is for api server. `core` is mostly shared by both modules. `automation` is for automation tests.
What do you suggest to use if not using sbt multi projects?

**Comment by Qi77Qi** [comment]
@dvoet , my original thought was, 
1. if we're making creating notifications and topics in rawls, we'll need to make more changes once we don't rely on rawls for uploading metadata. With current approach in this PR, all we need to change is who publishes metadata to the bucket later. I actually would've created the bucket in hamm too if that's easier.
2. For a given bucket, we can have multiple notifications. So it feels more intuitive to me that subscriber is responsible for creating notifications and topics these notifications go to, and give subscriber access to the bucket.

**Comment by dvoet** [comment]
if bucket is always going to be the way, which I think likely given the potential size of metadata, I think hamm should own the bucket creation.... and since you are moving it ;) you don't need to name it with the service account client id, call it hamm-cromwell-metadata-[env]... need to figure out what rawls in fiab will write to though.

File: build.sbt (hunk 1 above)
[the remark this answers is not in the document]
Comment (dvoet) [reply build.sbt]: I only mention it because the comment talks about core so I though if it were all in a single module maybe it wouldn't be a problem.

**Comment by dvoet** [comment]
in other words: the interface through which hamm gets metadata is dropping a metadata file in hamm's bucket. How hamm knows that there is a new file is an internal implementation detail of hamm.

so hamm owns the bucket and notification setup

File: build.sbt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Qi77Qi) [reply build.sbt]: I've seen other repos using `BuildInfoPlugin` just fine, and all of them have multi projects. I suspect this is due to protobuf module also has managed source generation, but somehow it messes up `BuildInfoPlugin` source generation.

**Comment by Qi77Qi** [comment]
@dvoet , I'm thinking leaving bucket creation in rawls for now until we figure out fiab for hamm and other deployment things, what do you think? If later we'll terraform resources (I think tf is a better way to go for managing resources, including service accounts. It'll be easier to apply permission updates etc too which is hard currently in app code, tf can detect any configuration updates and apply them properly), the code change will be deprecated too.

**Comment by Qi77Qi** [comment]
https://github.com/DataBiosphere/hamm/pull/9 is terraform code for creating service accounts and will address the two manual steps I mentioned in this PR as well

**Comment by dvoet** [comment]
I think we should move it. I don’t anticipate it moving to tf any time
soon. If fiab is a problem make writing the metadata optional. If a bucket
is not configured don’t write.

**Pull request closed without merging**